src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosJobFramework.scala

```diff
@@ -234,8 +234,17 @@ class MesosJobFramework @Inject()(
           log.info("Task with id '%s' FAILED".format(taskId))
           scheduler.handleFailedTask(taskStatus)
         case TaskState.TASK_LOST =>
-          log.info("Task with id '%s' LOST".format(taskId))
-          scheduler.handleFailedTask(taskStatus)
+          if (taskManager.persistenceStore.getTasks.contains(taskId)) {
+            //if we know about this, then it's the real deal and the task is lost. 
+            //we treat lost tasks as failed.
+            log.info("TASK_LOST for task %s. Treating as a failed run.".format(taskId))
```

Comment:
maybe, though I'm fairly sure we shouldn't ever get a TASK_FAILED after a TASK_FINISHED?